### PR TITLE
task-9368+9369+9370: transaction detail screen: create ui + handle event + fetch data + api

### PR DESCRIPTION
**Pull request by vanpd-0859**
### Issue Link :link:

https://edu-redmine.sun-asterisk.vn/issues/9368
https://edu-redmine.sun-asterisk.vn/issues/9369
https://edu-redmine.sun-asterisk.vn/issues/9370
### Goals :soccer:

Task 9368: 
https://marvelapp.com/3feje8d/screen/54847312
- Transaction type image
- Amount + value asset labels
- Recipient, transaction #, network fee, comfirmations, transaction time, nonce labels
- More details hyperlink label
- Share nav button
Task 9369:
- Click More details hyperlink label to go to etherscan.io on browser
- Click Share nav button to open Share controller
Task 9370:
Fetch data from API and display on UI:
- All informations on labels

### Implementation Details :construction:

https://gfycat.com/AdmiredAgreeableKoi

### CODE REVIEW

**Comment by vanpd-0859** [comment]
ready

File: CryptoWallet/Extensions/Date+.swift [hunk 1]
```diff
@@ -9,19 +9,19 @@
 import Foundation
 
 extension Date {
-    static func convertTimeStampToDate(timeStamp: Double) -> String {
+    static func convertTimeStampToDate(timeStamp: Double, format: String? = nil) -> String {
         let date = Date(timeIntervalSince1970: timeStamp)
         let dateFormatter = DateFormatter()
         dateFormatter.timeZone = TimeZone(abbreviation: "UTC+7")
-        dateFormatter.dateFormat = "MMM dd, yyyy"
+        dateFormatter.dateFormat = format == nil ? "MMM dd, yyyy" : format
```
Comment (longvt96) [inline CryptoWallet/Extensions/Date+.swift]:
```suggestion
        dateFormatter.dateFormat = format ?? "MMM dd, yyyy"
```

File: CryptoWallet/Extensions/Date+.swift [hunk 2]
```diff
@@ -9,19 +9,19 @@
...
         dateFormatter.locale = NSLocale.current
         return dateFormatter.string(from: date)
     }
     
-    static func convertDateToString(date: Date) -> String {
+    static func convertDateToString(date: Date, format: String? = nil) -> String {
         let dateFormatter = DateFormatter()
         dateFormatter.timeZone = TimeZone(abbreviation: "UTC+7")
-        dateFormatter.dateFormat = "MMM dd, yyyy"
+        dateFormatter.dateFormat = format == nil ? "MMM dd, yyyy" : format
```
Comment (longvt96) [inline CryptoWallet/Extensions/Date+.swift]:
```suggestion
        dateFormatter.dateFormat = format ?? "MMM dd, yyyy" 
```

File: CryptoWallet/Scenes/Home/Wallet/Asset/TransactionDetail/TransactionDetailHeader.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+//
+//  Created by Phan Dinh Van on 4/19/19.
+//  Copyright © 2019 Phan Dinh Van. All rights reserved.
+//
+
+import UIKit
+import Reusable
+
+final class TransactionDetailHeader: UIView, NibOwnerLoadable {
+    @IBOutlet weak var transactionTypeImage: UIImageView!
```
Comment (longvt96) [inline CryptoWallet/Scenes/Home/Wallet/Asset/TransactionDetail/TransactionDetailHeader.swift]: add private

File: CryptoWallet/Scenes/Home/Wallet/Asset/TransactionDetail/TransactionDetailHeader.swift [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    @IBOutlet weak var amountLabel: UILabel!
```
Comment (longvt96) [inline CryptoWallet/Scenes/Home/Wallet/Asset/TransactionDetail/TransactionDetailHeader.swift]: tương tự

**Comment by vanpd-0859** [comment]
ready

**Review by longvt96 — APPROVED** [review]

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**